src/solvers.jl

```diff
@@ -791,49 +725,108 @@ function conicdata(m::Model)
     if !isempty(eq_rows)
         push!(con_cones, (:Zero,eq_rows))
     end
-    @assert c == numLinRows + numBounds
+    c
+end
 
+function fillconstrRHS!(b, con_cones, c, socconstr::Vector{SOCConstraint})
+    for con in socconstr
+        expr = con.normexpr
+        c += 1
+        soc_start = c
+        b[c] = -expr.aff.constant
+        for term in expr.norm.terms
+            c += 1
+            b[c] = expr.coeff*term.constant
+        end
+        push!(con_cones, (:SOC, soc_start:c))
+    end
+    c
+end
+
+# Represents the constraints `constr` as
+# `b - Ax ∈ K`,
+# writes the matrix `A` of this representation in the sparse format using `I`, `J` and `V`,
+# starting at row index `c+1`, and specifies the indices of each constraint in `constr_dual_map`,
```

Comment:
I agree the name can be improved (a quick git blame show that the name was coined by @emreyamangil).
`c` is the current row in the matrix while `d` is the current index in the vector of constraints stored in the JuMP model. So for example a SOC constraint increments `d` only once while it intcrements `c` `n+1` times.